database/sinkdb/state.go

```diff
@@ -92,6 +92,13 @@ func (s *state) RestoreSnapshot(data []byte, index uint64) error {
 	s.peers = snapshot.Peers
 	s.state = snapshot.State
 	s.version = snapshot.Version
+
+	// Chain Core 1.2.x generates snapshots without populating versions,
+	// so we manually initialize here to avoid nil errors after marshaling
+	// and unmarshaling the snapshot
```

Comment:
nit, but I'd add `, which will turn an initialized-but-empty map into a nil map`  to this comment.